### PR TITLE
Rnd settings

### PR DESCRIPTION
**Pull request by jburel**
This PR fixes a synchronization issue reported on the following card: https://trello.com/c/4mDNygGv/38-follow-up

To test:
 1. Log as user-6. Select the read-write group
 * Add user-5 to the display. 
 * Select the first image in the ```archived dv``` dataset under user-5
 * Go to preview tab
 * The image has been rnd settings for both user-5 and user-6
 * Select user-5 settings.
 * Click ```Launch Full Viewer```
 * Display the rendering controls (click on the left icon in the toolbar to display it)
 * Check that the thumbnail under user settings corresponding to user-5 is selected in the main viewer (it was user-6 w/o this PR)
 * Close the viewer w/o saving the settings.
 * Check that the settings are reset in the preview and the original thumbnail under User settings is selected i.e. user-6

Repeat the same test but with the following variation on step 7
  * one test with ```Click on Full Viewer``` in the general pane.
 * one test with ```Right click on the selected image in the tree```
 * one test with ```Right click on the selected image in the central pane```

cc @dominikl

### CODE REVIEW

**Comment by gusferguson** [comment]
@jburel @dominikl 

Tested using OMERO.insight-5.2.0-m1-227-df1a851-ice35-b114-mac eel user-6 read-write-1

Thumbnail in centre pane not updating correctly when selecting user-5 rendering settings as laid out in test scenario. Full viewer does show correct user-5 rendering settings when opened, but thumbnail remains as user-6 rendering settings.

user-6 user-6 connected to eel openmicroscopy org

main

**Comment by jburel** [comment]
@gusferguson: reading again the description
this is not clear enough.

**Comment by gusferguson** [comment]
@jburel 

Retested with revised testing description.

Note; centre pane thumbnail not expected to update unless new selection in Preview pane saved.
In Full Viewer with rendering settings panel open:
Full Viewer rendering settings thumbnail behaves as expected.
If settings selection is changed in Full Viewer from user-5 to user-6 - rendering settings change is reflected in Full Viewer window.
If change of selection not saved on closing Full Viewer window, there is inconsistent behaviour of panel view and thumbnail selected in Preview pane.
Usually Preview panel reverts to user-6 settings, but blue "selected" highlight does not change from user-5 thumbnail to user-6 thumbnail in Preview pane. 
This is intermittent though as it did change on one occassion.

**Comment by dominikl** [comment]
Just a note: The selection highlight around the user rendering settings is misleading in any case, as you always actually see your own rendering settings. E. g. selecting another user's settings and starting to modify, probably misleads the user to think he's editing the other user's settings but he always edits his own settings (selecting another user renderings settings really just changes his own rendering settings to match the other user's one, that's why you then can acutally save them as your own); if I remember right, that was also a point you mentioned on my original PR. Unfortunately I can't think of any solution which would not mislead the user in some way.

**Comment by jburel** [comment]
I see myself the "highlight" as the starting point of the changes for the user

**Comment by gusferguson** [comment]
@dominikl - agreed - will remain confusing - probably unavoidably - until we have multiple settings per user - which incidentally seems to have been kicked into the long grass.

**Comment by jburel** [comment]
@dominikl @gusferguson: The last commit should reset the selection in the preview when the viewer is closed.

**Comment by jburel** [comment]
@pwalczysko: The synchronization issue described in https://trac.openmicroscopy.org/ome/ticket/13025 should now be fixed

**Comment by jburel** [comment]
To test:
 * Redo the updated test from description

Also test the 2 workflow described in https://trac.openmicroscopy.org/ome/ticket/13025
 In that case, no need to view the image in preview first. you can open the image directly in the main viewer.
The out of synch issues occur in the main viewer.

**Comment by gusferguson** [comment]
@jburel @pwalczysko 

Tested using OMERO.insight-5.2.0-m1-270-5057004-ice35-b115-win eel logged in as user-6 and user-5 data read-write-1

Testing as described in PR:
- behaves as described
- however not convinced this is a logical workflow, but if this is what is wanted then this PR does it.

From this POV - good to merge.

Testing first workflow from ticket:
"In the Insight viewer rendering panel there is the option to select someone else's render settings. This is very useful, but the channel selection buttons to the left of the image do not update if there is a colour change (see attachment). This is very confusing when you try to inspect the image to see if you find the alternate render setting useful."
- appears to be resolved
- buttons to left change colour as expected.

From this POV - good to merge.

Testing first workflow from ticket:
"Open a multi-channel image with all channes visible (eg 9412 eel merge user-4).
Tick the Greyscale tickbox.
Observe that in the Rnd pane only the first channel is selected (the blue one) and the blue button is pressed down.
Observe that in the main viewer, not only the blue channel, but also the green and red channel buttons are depressed -> desynchronisation."
- only blue button is depressed in both the rendering and the main panel.

From this POV - good to merge.

**Comment by joshmoore** [comment]
@dominikl : code changes ok in terms of other ongoing PRs?

File: components/insight/SRC/org/openmicroscopy/shoola/agents/imviewer/view/ImViewerFactory.java [hunk 1]
```diff
@@ -287,6 +287,23 @@ public static void rndSettingsSaved(long pixelsID, RndProxyDef settings)
 	}
 
 	/**
+	 * Indicates that rendering settings have been modified.
+	 * 
+	 * @param ImageID The Identifier of the pixels set.
```
Comment (sbesson) [inline components/insight/SRC/org/openmicroscopy/shoola/agents/imviewer/view/ImViewerFactory.java]: Capitalization issue  - see https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-build/115/warnings39Result/

**Comment by dominikl** [comment]
Looks fine! Should have done all this `AgentEvent` based from the beginning, would have avoided quite some problems... 
But the issue mentioned in the description doesn't work for me with todays build. I select user-5 settings in preview, then open full viewer. The full viewer shows user-5 settings, but the rendering settings panel (of the full viewer) shows user-6 settings highlighted.

**Comment by jburel** [comment]
@dominikl: did you use the image in the description?

**Comment by pwalczysko** [comment]
@jburel : I can confirm @dominikl 's observations. I am using the image in the description of this PR (ID ID: 9506). Although the rendering setting shown in the Full Viewer itself are the expected ones (= user-5's settings), the Rnd pane in Full viewer shows still the Rnd of user-6 as selected.

**Comment by pwalczysko** [comment]
@jburel : On the other hand, all the problems reported by @mporter-gre are fixed here now - the sync between the Rnd pane in full viewer and the channel buttons in full viewer is expected for both Grayscale and change Rnd settings cases.

**Comment by jburel** [comment]
one more synchronization issue noticed if the user this time launches the full viewer from the "General Tab" after selecting the settings of user-5 in the preview tab.
I will look at that one now

**Comment by jburel** [comment]
The various synchronization issues are now fixed when launching the full viewer.
To test the last view commits.
Redo the test in the description, I have added few variations from the initial description

**Comment by dominikl** [comment]
Works fine! Only thing I noticed is that the fullviewer moves into the background directly after opening, if it's launched via tree or central panel and another user's rendering setting has been selected in the preview. But in my opinion that's negligible.

**Comment by jburel** [comment]
@dominikl: I don't think that is related to this PR.

**Comment by dominikl** [comment]
That's true, sorry. It also happens with the latest build, just never noticed that before.

**Comment by pwalczysko** [comment]
Works fine for me as well, sorry for being late here.

**Pull request merged**